fbw-a380x/src/systems/extras-host/modules/flt_sync/LoadFltStateManager.ts

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2022 FlyByWire Simulations
+// SPDX-License-Identifier: GPL-3.0
+
+import { EventBus } from '@microsoft/msfs-sdk';
+import { MfdSurvEvents } from '../../../instruments/src/MsfsAvionicsCommon/providers/MfdSurvPublisher';
+import { TcasMode } from '../../../systems-host/systems/tcas/lib/TcasConstants';
+
+/**
+ * This class is used to sync event consumers when starting from a spawn state (Gate/Cold and Dark vs Runway/Approach/Cruise)
+ */
+
+enum FltState {
+  Hanger = 1,
+  Apron = 2,
+  Taxi = 3,
+  Runway = 4,
+  Climb = 5,
+  Cruise = 6,
+  Approach = 7,
+  Final = 8,
+}
+export class LoadFltStateManager {
+  constructor(private readonly bus: EventBus) {
+    console.log('LoadFlt: Created');
+  }
+
+  public connectedCallback(): void {
+    // empty
+  }
+
+  public startPublish(): void {
+    console.log('LoadFlt: startPublish()');
+
+    switch (SimVar.GetSimVarValue('L:A32NX_START_STATE', 'Enum')) {
+      case FltState.Hanger:
+      case FltState.Apron:
+        break;
+      case FltState.Taxi:
+      case FltState.Runway:
+      case FltState.Climb:
+      case FltState.Cruise:
+      case FltState.Approach:
+      case FltState.Final:
+      default:
+        this.bus.getPublisher<MfdSurvEvents>().pub('mfd_xpdr_set_auto', true, true);
+        this.bus.getPublisher<MfdSurvEvents>().pub('tcas_alert_level', TcasMode.TARA, true);
+        break;
+    }
```

Comment:
It would be a lot simpler to let each subsytem do this itself rather than publishing bus events from a random location.

Reply:
I've now moved it into Transponder.ts and LegacyTcasComputer.ts, but, this still isn't perfect. It's not immediately obvious to other devs that this event is being manipulated based on spawn state if you are working on a subsystem that is dependent on the XPDR and TCAS computer (Like the MFD SURV page).

Note that If we wish to support .FLT loading from saves and more advanced state management in the future, it would make sense to revisit this then.